File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/matcher/ClassTypeMatcherInterface.java
package com.github.tobi812.sprykerplugin.models.matcher;

import com.intellij.psi.PsiDirectory;
import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;

public interface ClassTypeMatcherInterface {

    ClassDefinitionInterface matchClassType(String fqClassName);

    Boolean classTypeMatchesDir(String classType, PsiDirectory directory) throws Exception;

    String matchProjectName(String classType, PsiDirectory psiDirectory) throws Exception;

    String matchProjectName(String fqClassName);

    String matchBundleName(String classType, PsiDirectory psiDirectory) throws Exception;

    String matchBundleNameFromFQName(ClassDefinitionInterface classDefinition, String fqName);

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/matcher/ClassTypeMatcherInterface.kt
package com.github.tobi812.sprykerplugin.models.matcher

import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface
import com.intellij.psi.PsiDirectory
import java.lang.Exception

interface ClassTypeMatcherInterface {
    fun matchClassType(fqClassName: String): ClassDefinitionInterface?

    @Throws(Exception::class)
    fun classTypeMatchesDir(classType: String, directory: PsiDirectory): Boolean

    @Throws(Exception::class)
    fun matchProjectName(classType: String, psiDirectory: PsiDirectory): String
    fun matchProjectName(fqClassName: String): String

    @Throws(Exception::class)
    fun matchBundleName(classType: String, psiDirectory: PsiDirectory): String
    fun matchBundleNameFromFQName(classDefinition: ClassDefinitionInterface, fqName: String): String
}
